usr.sbin/bhyve/bhyverun.c

```diff
@@ -1356,7 +1365,25 @@ main(int argc, char *argv[])
 			exit(1);
 		}
 	}
-#endif
+
+	if (receive_migration != NULL) {
+		if (vm_pause_user_devs(ctx) != 0) {
+			fprintf(stderr, "Failed to pause PCI device state.\n");
+			exit(1);
+		}
+
+		fprintf(stdout, "Starting the migration process...\r\n");
+		if (receive_vm_migration(ctx, receive_migration) != 0) {
+			fprintf(stderr, "Failed to migrate the vm.\r\n");
+			exit(1);
+		}
```

Comment:
I do not know the final format of the migration feature (whether it will be accepted under the BHVYE_SNAPSHOT or if we should add another define for the migration - e.g. BHYVE_WARM_MIGRATION as to activate this feature) and I would like to let it as it is. 

I've noted the observation related to the three restore functions. We should have a separate review for that. I do not want to interfere with the warm migration patch.